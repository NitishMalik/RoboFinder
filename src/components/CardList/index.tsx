import React from "react";
import Card from "../Card";

const CardList = ({ robos }) => {
  console.log("Render Card List");
  return (
    <div>
      {robos.map((robo) => {
        return (
          <Card
            key={robo.id}
            id={robo.id}
            name={robo.name}
            email={robo.email}
          />
        );
      })}
    </div>
  );
};

export default CardList;
