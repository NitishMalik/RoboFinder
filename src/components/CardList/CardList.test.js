import { shallow } from "enzyme";
import React from "react";
import CardList from "./";
//import toJson from "enzyme-to-json";

test("should render CardList component", () => {
  const mockRobos = [
    {
      id: 1,
      name: "John Snow",
      email: "john@gmail.com",
    },
  ];
  expect(shallow(<CardList robos={mockRobos} />).debug()).toMatchSnapshot();
});
