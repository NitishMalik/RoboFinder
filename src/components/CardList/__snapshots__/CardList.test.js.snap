// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CardList component 1`] = `
"<div>
  <Card id={1} name=\\"John Snow\\" email=\\"john@gmail.com\\" />
</div>"
`;
